data-raw/example_plots.R

```diff
@@ -0,0 +1,155 @@
+         z = sample(1:10, 20, replace = TRUE))
+
+
+  df_scatter %>%
+    ggplot(aes(x, y, size = z, color = y > 100)) +
+    geom_point() +
+    labs(x = NULL, y = NULL,
+         title = "TITLE",
+         subtitle = "caption/description",
+         caption = "data source") +
+    si_style() +
+    theme(legend.position = "none")
+
+
+
+# LINE CHART --------------------------------------------------------------
+
+
+  df_line <- cascade %>%
+    filter(indicator == "TX_NEW",
+           period_type == "results") %>%
+    mutate(pt_start = case_when(period == min(period) ~ value),
+           pt_end = case_when(period == max(period) ~ value),
+           partner_focal = primepartner == "Sagittarius") %>%
+    arrange(primepartner, period)
+
+  largest_partners <- df_line %>%
+    top_n(5, wt = pt_end) %>%
+    pull(primepartner)
+
+  df_line <- filter(df_line, primepartner %in% largest_partners)
+
+
+
+  df_line %>%
+    ggplot(aes(period, value, group = primepartner, color = partner_focal)) +
+    geom_path(size = 1) +
+    geom_point(aes(y = pt_end), shape = 21, size = 4, stroke = 1.5, fill = "white", na.rm = TRUE) +
+    geom_point(aes(y = pt_start, fill = partner_focal), shape = 21, size = 4, na.rm = TRUE) +
+    geom_text(aes(y = pt_end, label = primepartner), na.rm = TRUE,
+              family = "Source Sans Pro", nudge_x = .4) +
+    labs(x = NULL, y = NULL,
+         title = "TITLE",
+         subtitle = "caption/description",
+         caption = "data source") +
+    scale_x_discrete(expand = expansion(add = 1)) +
+    si_style() +
+    theme(legend.position = "none")
+
+
+# BAR CHART ---------------------------------------------------------------
+
+
+  df_bar <- cascade %>%
+    filter(operatingunit == "Saturn",
+           indicator == "TX_NEW",
+           period == "FY50") %>%
+    group_by(primepartner, period, period_type) %>%
+    summarise(value = sum(value, na.rm = TRUE)) %>%
+    ungroup() %>%
+    spread(period_type, value) %>%
+    mutate(targets_q1 = targets * .25) %>%
+    filter(!is.na(targets_q1))
+
+  set.seed(23)
+
+  agency <-
+    tibble(fundingagency = sample(c("USAID", "CDC"), nrow(df_bar), replace = TRUE))
+
+  df_bar <- df_bar %>%
+    bind_cols(agency, .)
+
+  df_bar %>%
+    ggplot(aes(fct_reorder(primepartner, targets_q1, sum), cumulative)) +
+    geom_col(aes(y = targets_q1), fill = "gray90") +
+    geom_col() +
+    geom_errorbar(aes(ymin = targets_q1, ymax = targets_q1), color = "gray90", size = .9) +
+    scale_y_continuous(label = comma, expand = c(.005, .005)) +
+    facet_wrap(~ fct_rev(fundingagency), scales = "free_y") +
+    coord_flip() +
+    labs(x = NULL, y = NULL,
+         title = "TITLE",
+         subtitle = "caption/description",
+         caption = "data source") +
+    si_style_xgrid()
+
+
+
+# LOLLIPOP CHART ----------------------------------------------------------
+
+
+  df_lollipop <- df_bar %>%
+    mutate(achievement = cumulative / targets)
+
+  df_lollipop %>%
```

Comment:
would it make more sense to use tidytext::reorder_within() to have the partners ordered within the Agencies?

`df_lollipop %>%
    mutate(partner_order = tidytext::reorder_within(primepartner, achievement, agency)) %>%   
    ggplot(aes(partner_order, achievement)) + 
    geom_segment(aes(x = partner_order, y = 0, xend = partner_order, yend = achievement), size = .9) +  
    geom_point(size = 3) +
    tidytext::scale_x_reordered() +
  ...`

Reply:
sounds pretty ideal to me :)